jerry-core/ecma/base/ecma-helpers-errol.c

```diff
@@ -230,6 +230,113 @@ ecma_errol0_dtoa (double val, /**< ecma number */
 } /* ecma_errol0_dtoa */
 
 /**
+ * Calculate the number of digits from the given double value which contains no franction part
+ *
+ * @return number of digits
+ */
+inline static int32_t __attr_always_inline___
+ecma_number_of_digits (double val) /**< ecma number */
+{
+  JERRY_ASSERT (val > 0.0);
+  JERRY_ASSERT (fabs (fmod (val, 1.0)) < ERROL0_EPSILON);
+  int32_t exponent = 0;
+
+  while (val >= 1.0)
+  {
+    val /= 10.0;
+    exponent++;
+  }
+
+  return exponent;
+} /* ecma_number_of_digits */
+
+inline static void __attr_always_inline___
+ecma_double_to_ascii (double val, /**< ecma number */
+                     lit_utf8_byte_t *buffer_p, /**< buffer to generate digits into */
+                     int32_t *exp_p) /**< [out] exponent */
+{
+  int32_t char_cnt = 0;
+  int32_t num_of_digits = ecma_number_of_digits (val);
+
+  double divider = 10.0;
+  double prev_residual;
+  double mod_res = fmod (val, divider);
+
+  buffer_p[num_of_digits - 1 - char_cnt++] = (lit_utf8_byte_t) ((int) mod_res + '0');
+  divider *= 10.0;
+  prev_residual = mod_res;
+
+  while (char_cnt < num_of_digits)
+  {
+    mod_res = fmod (val, divider);
+    double residual = mod_res - prev_residual;
+    buffer_p[num_of_digits - 1 - char_cnt++] = (lit_utf8_byte_t) ((int) (residual / (divider / 10.0)) + '0');
+
+    divider *= 10.0;
+    prev_residual = mod_res;
+  }
+
+  *exp_p = char_cnt;
+
+} /* ecma_double_to_ascii */
+
+/**
+ * Errol0 double to binary floating-point number conversion
+ *
+ * @return number of generated digits
+ */
+inline lit_utf8_size_t __attr_always_inline___
+ecma_errol0_dtobfpn (double val, /**< ecma number */
+                     lit_utf8_byte_t *buffer_p, /**< buffer to generate digits into */
+                     int32_t *exp_p) /**< [out] exponent */
+{
+  lit_utf8_byte_t integer_part_buffer[325];
```

Comment:
Good catch. Much less is enough! IEEE 754 double precision floating point range is beetween ± ≈10^−323.3 to ≈10^308.3